config/config_test.go

```diff
@@ -0,0 +1,210 @@
+package config_test
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"testing"
+
+	"github.com/ONSdigital/log.go/v3/config"
+	"github.com/ONSdigital/log.go/v3/log"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestFromEnv(t *testing.T) {
+	Convey("Starting with a clear environment", t, func() {
+		os.Clearenv()
+
+		Convey("No vars leaves config unchanged", func() {
+			wanted := config.Config{Namespace: "some namespace"}
+			cfg := wanted
+			config.FromEnv(&cfg)
+			So(cfg, ShouldEqual, wanted)
+		})
+
+		Convey("Setting HUMAN_LOG updates pretty config", func() {
+			cfg := config.Config{}
+
+			tests := []struct {
+				value  string
+				wanted bool
+			}{
+				{"1", true},
+				{"0", false},
+				{"t", true},
+				{"f", false},
+				{"true", true},
+				{"false", false},
+			}
+
+			for _, test := range tests {
+				Convey("When HUMAN_LOG="+test.value, func() {
+					os.Setenv("HUMAN_LOG", test.value)
+					config.FromEnv(&cfg)
+					So(cfg.Pretty, ShouldEqual, test.wanted)
+				})
+			}
+		})
+
+		Convey("Setting LOG_LEVEL updates level in config", func() {
+			initialLevel := slog.Level(327)
+			cfg := config.Config{Level: initialLevel}
+
+			tests := []struct {
+				value  string
+				wanted slog.Level
+			}{
+				{"DEBUG", log.LevelDebug},
+				{"debug", log.LevelDebug},
+				{"Debug", log.LevelDebug},
+				{"-4", log.LevelDebug},
+
+				{"INFO", log.LevelInfo},
+				{"info", log.LevelInfo},
+				{"Info", log.LevelInfo},
+				{"0", log.LevelInfo},
+
+				{"WARN", log.LevelWarn},
+				{"warn", log.LevelWarn},
+				{"Warn", log.LevelWarn},
+				{"4", log.LevelWarn},
+
+				{"ERROR", log.LevelError},
+				{"error", log.LevelError},
+				{"Error", log.LevelError},
+				{"8", log.LevelError},
+
+				{"FATAL", log.LevelFatal},
+				{"fatal", log.LevelFatal},
+				{"Fatal", log.LevelFatal},
+				{"12", log.LevelFatal},
+
+				{"99", slog.Level(99)},
+
+				// Invalid config leaves level in config unchanged
+				{"nonsense", initialLevel},
+			}
+
+			for _, test := range tests {
+				Convey("When HUMAN_LOG="+test.value, func() {
+					cfg.Level = initialLevel
+					os.Setenv("LOG_LEVEL", test.value)
```

Comment:
There's an unhandled error returned by Setenv here too.

Reply:
Thank you, addressed in https://github.com/ONSdigital/log.go/pull/44/commits/fc70a349237c838e973a6e61addef0ee83910665